app/controllers/users_controller.rb

```diff
@@ -19,24 +19,13 @@ def new_with_invitation
 
   def create
     @user = User.new(user_params)
+    result = UserSignup.new(@user).sign_up(stripe_token: params[:stripeToken], invitation_token: params[:invitation_token])
 
-    begin
-      if @user.valid?
-        create_stripe_customer
-        charge_stripe_customer
-
-        @user.save
-        handle_invitation
-        flash[:success] = "Welcome, #{@user.name}! Your account has been created, please login below."
-        UserMailer.delay.welcome(@user.id)
-
-        redirect_to login_path
-      else
-        render 'new'
-      end
-
-    rescue Stripe::CardError => e
-      flash[:danger] = e.message
+    if result.succesful?
+      flash[:success] = "Welcome, #{@user.name}! Your account has been created, please login below."
+      redirect_to login_path
+    else
+      flash[:danger] = result.error_message
```

Comment:
When you render, then flash.now is more suitable. You dont want to show the error on next GET request.